cluster-autoscaler/processors/podinjection/pod_injection_processor.go

```diff
@@ -60,10 +60,17 @@ func (p *PodInjectionPodListProcessor) Process(ctx *context.AutoscalingContext,
 		return unschedulablePods, fmt.Errorf("failed to list nodeInfos from cluster snapshot: %v", err)
 	}
 	scheduledPods := podsFromNodeInfos(nodeInfos)
-
 	groupedPods := groupPods(append(scheduledPods, unschedulablePods...), controllers)
-	var podsToInject []*apiv1.Pod
 
+	allPods, err := ctx.AllPodLister().List()
```

Comment:
Why list all pods? You can decide if a pod has a scheduling gate or not just by looking at it, no need to cross reference the list of pods with itself I think?